### PR TITLE
Task/null value in parse step

### PR DESCRIPTION
**Pull request by kzangeli**
'bool typeGiven' in Entity, ContextAttribute and Metadata.
If typeGiven == true, then the 'type' was present during the JSON parse of the payload.

### CODE REVIEW

File: src/lib/apiTypesV2/Entity.cpp [hunk 1]
```diff
@@ -38,6 +38,7 @@
 */
 Entity::Entity()
 {
+  typeGiven = false;
```
Comment (fgalan) [inline src/lib/apiTypesV2/Entity.cpp]: Actually, the title of the PR should be "null type in parse step", right? I mean, the ability from the parser layer to distinguish between the situation in which the input payload includes `type: ""` (typeGiven = false) and absences of `type` (typeGiven = true).

Just to clarify this PR is not about the usage of `null` as value for an attribute, as boolean or numbers are.

Comment (fgalan) [inline src/lib/apiTypesV2/Entity.cpp]: NTC (it seems pretty clear reading the code and its comments)

**Comment by fgalan** [comment]
LGTM

File: src/lib/apiTypesV2/Entity.cpp (hunk 1 above)
Comment (kzangeli) [inline src/lib/apiTypesV2/Entity.cpp]: Yes, of course, the title is poorly chosen.

However, a clarification to your comment:

 type: "" (typeGiven = **true**)
absences of type (typeGiven = **false**)

Comment (fgalan) [inline src/lib/apiTypesV2/Entity.cpp]: You are right, It was a typo... I tried meaning just the opossite :$

**Pull request merged**